### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by lucasiragusa** [lucasiragusa/landingpermit_app#20]

**Comment by Zaubeerer** [comment]
No, let's have a look at the code together, I may already have some quick ideas :) 

Can you create a PR in which you add a `# TODO` comment in the respective files? 

Then, I can already make some educated guesses.

**Comment by Zaubeerer** [comment]
As discussed in our call, you have different options to analyze the performance of your code:
1. run `pytest . --durations=10`, showing you the 10 slowest tests
2. Use cProfile or @profile decorator as demonstrated by Bob

**Comment by lucasiragusa** [comment]
I wasn't able to write a test for the function that I really want to test, because if you remember it isn't a function with input/output but more like a script "dressed" as a function. 

So, what I did instead is run the script with a profiler: 

Example: 

`
lucasiragusa@Lucas-MacBook-Pro` src % python3 -m cProfile -s time main.py ../data/ssim/EY_SSIM_small.ssim > profiling_results.txt  
`

Based on the first results, I refactored the code a bit to change the way that the table was being created and avoid the use of .cells (as suggested here: https://github.com/python-openxml/python-docx/issues/174). 

This has somewhat improved run time but the performance remains very slow (~2mins for a 500 lines SSIM and infeasibly slow for 12k lines SSIM). 

I am attaching the results of the profiling. While I am not 100% sure how to interpret these results, I read that the functions that are taking the most times seem to be buried deep inside python-docx, something to do with xml reading. 

[profiling_results_small.txt](https://github.com/lucasiragusa/landingpermit_app/files/13865103/profiling_results_small.txt)

Now the question I have is, I will commit this additional refactoring before merging.

**Comment by Zaubeerer** [comment]
OK, I suggest to first focus on making all the other things work, as 2 mins for 500 lines SSIM seems OK for a prototype/MVP.

We can always improve performance when all the rest works:
1. Make it work
2. Make it right
3. Make it fast

**Comment by lucasiragusa** [comment]
Perfectly agreed on Make it work, Make it right, Make it fast. 

I will leave the issue open nevertheless so it remains tracked
